pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -31,16 +32,43 @@ type StepResolver interface {
 }
 
 type OperatorStepResolver struct {
-	subLister              v1alpha1listers.SubscriptionLister
-	csvLister              v1alpha1listers.ClusterServiceVersionLister
-	client                 versioned.Interface
-	globalCatalogNamespace string
-	resolver               *Resolver
-	log                    logrus.FieldLogger
+	subLister                 v1alpha1listers.SubscriptionLister
+	csvLister                 v1alpha1listers.ClusterServiceVersionLister
+	client                    versioned.Interface
+	globalCatalogNamespace    string
+	resolver                  *Resolver
+	log                       logrus.FieldLogger
+	globalCatalogSourceToggle GlobalNamespaceToggle
 }
 
 var _ StepResolver = &OperatorStepResolver{}
 
+type GlobalNamespaceToggle interface {
+	ShouldEnableGlobalNamespaceResolution(ctx context.Context, namespace string) (bool, error)
+}
+
+func NewNamespaceAnnotationGlobalCatalogToggle(opClient operatorclient.ClientInterface) *NamespaceAnnotationGlobalCatalogToggle {
+	return &NamespaceAnnotationGlobalCatalogToggle{
+		opClient: opClient,
+	}
+}
+
+type NamespaceAnnotationGlobalCatalogToggle struct {
+	opClient operatorclient.ClientInterface
```

Comment:
Catalog operator should already have an informer for namespaces -- you likely want to query via the Lister interface instead of directly making an API request.